### ISSUE TITLE
BAM support

### ISSUE THREAD

**Issue by epaule** [GMOD/Apollo#1868]
Hia,

I was trying out the new master (with the new JBrowse) and when clicking on "create new annotation" -> "gene" on a BAM feature

I get: 
Error Performing Operation
Cannot get property 'name' on null object

converting into other types works.

but the feature technically has a name:
![image](https://user-images.githubusercontent.com/20523/37090595-e1603a9e-21fc-11e8-924e-239e37435f6c.png)

tracklist bit:
```
      {
         "key" : "Full transcript RNASeq data",
         "label" : "isoseq_all",
         "storeClass" : "JBrowse/Store/SeqFeature/BAM",
         "type" : "JBrowse/View/Track/Alignments2",
         "urlTemplate" : "bam/TTRE_all_isoseq.bam",
        "default_biotype" : "gene"
      },
```

error log:
```
2018-03-07 11:36:28,712 [Actor Thread 9] ERROR apollo.FeatureService  - CV Term not known match for CV sequence
2018-03-07 11:36:28,712 [Actor Thread 9] WARN  apollo.FeatureService  - Feature type not set for [cv:[name:sequence], name:match]
2018-03-07 11:36:28,712 [Actor Thread 9] ERROR apollo.AnnotationEditorController  - CAUGHT ERROR through websocket call: java.lang.reflect.InvocationTargetException
2018-03-07 11:36:28,712 [Actor Thread 9] ERROR apollo.AnnotationEditorController  - THROWING PARENT ERROR instead through reflection: java.lang.NullPointerException: Cannot get property 'name' on null object
2018-03-07 11:36:28,712 [Actor Thread 9] ERROR apollo.AnnotationEditorController  - exception java.lang.NullPointerException: Cannot get property 'name' on null object
2018-03-07 11:36:28,712 [Actor Thread 9] ERROR apollo.AnnotationEditorController  - exception message Cannot get property 'name' on null object
2018-03-07 11:36:28,712 [Actor Thread 9] ERROR apollo.AnnotationEditorController  - username admin@local.host
2018-03-07 11:36:28,712 [Actor Thread 9] ERROR apollo.AnnotationEditorController  - error destination message: /topic/AnnotationNotification/user/admin@local.host
``` 
console log:
```
selection selection
AnnotTrack.js:1152 process: 1
AnnotTrack.js:5981 connected and sending notifications
AnnotTrack.js:5983 sent notification message
AnnotTrack.js:5981 connected and sending notifications
AnnotTrack.js:5983 sent notification message
```

**Comment by nathandunn** [comment]
Thanks for the detailed log.   @deepakunni3  Do you think this might be related to:

https://github.com/GMOD/Apollo/pull/1864 . ? 

I'll take a look at it.

**Comment by nathandunn** [comment]
Notes based on https://github.com/GMOD/Apollo/commits/master


### not working
- d50dcf56c5c765ca52a2b2d0fa0742ddf63b15e4
- e886efc9b66c0d039e8dd1f3d8cde357dbc9f359


### likely culprit
- #1863 (default biotype fix for #1861)
- https://github.com/GMOD/Apollo/commit/be73f3032fe99aaeb3f389056bd7944bccee0491

### working
- f46de7f8d0bf69f68901ba31e88f6a13b5b93019
- 2.0.8
